core/storage/app_feedback_report/gae_models.py

```diff
@@ -0,0 +1,659 @@
+                'platform_version': report_model.platform_version,
+                'device_country_locale_code': (
+                    report_model.device_country_locale_code),
+                'android_device_model': report_model.android_device_model,
+                'android_sdk_version': report_model.android_sdk_version,
+                'entry_point': report_model.entry_point,
+                'entry_point_topic_id': report_model.entry_point_story_id,
+                'entry_point_story_id': report_model.entry_point_story_id,
+                'entry_point_exploration_id': (
+                    report_model.entry_point_exploration_id),
+                'entry_point_subtopic_id': report_model.entry_point_subtopic_id,
+                'text_language_code': report_model.text_language_code,
+                'audio_language_code': report_model.audio_language_code,
+                'android_report_info': report_model.android_report_info,
+                'android_report_info_schema_version': (
+                    report_model.android_report_info_schema_version),
+                'web_report_info': report_model.web_report_info,
+                'web_report_info_schema_version': (
+                    report_model.web_report_info_schema_version)
+            }
+        return user_data
+
+    @staticmethod
+    def get_model_association_to_user():
+        """Model is exported as multiple instances per user since there
+        are multiple reports relevant to a user.
+        """
+        return base_models.MODEL_ASSOCIATION_TO_USER.MULTIPLE_INSTANCES_PER_USER
+
+    @staticmethod
+    def get_lowest_supported_role():
+        """The lowest supported role for feedback reports will be moderator."""
+        return feconf.ROLE_ID_MODERATOR
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id):
+        """Check whether AppFeedbackReportModel exists for user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether a model is associated with the user.
+        """
+        return cls.query(
+            cls.scrubbed_by == user_id).get(keys_only=True) is not None
+
+
+class AppFeedbackReportTicketModel(base_models.BaseModel):
+    """Model for storing tickets created to triage feedback reports.
+
+    Instances of this model contain information about ticket and associated
+    reports.
+
+    The id of each model instance is created by combining the entity's
+    ticket_name hash, creation timestamp, and a random 16-character string.
+    """
+
+    # A name for the ticket given by the maintainer, limited to 100 characters.
+    # Tickets with the same ID must have the same name.
+    ticket_name = datastore_services.StringProperty(required=True, indexed=True)
+    # The Github issue number that applies to this ticket.
+    github_issue_number = datastore_services.IntegerProperty(
+        default=None, indexed=True)
```

Comment:
Don't think you need to explicitly specify default=None.

Reply:
Ack, I was trying to think what would be best since if we want this to be indexable then it must be required=True, but when we first create a ticket there would be no Github issue.

Will remove default and specify None in create(), which I think is what it currently does already.

Reply:
I don't think indexable implies required=True... I've seen None values in indexed fields before.

Reply:
Huh you're right although I did see an error on it but that might have been due to a String / TextProperty.

Will remove the default